src/main/java/org/opensearch/neuralsearch/query/NeuralSparseQueryBuilder.java

```diff
@@ -95,20 +96,30 @@ public NeuralSparseQueryBuilder(StreamInput in) throws IOException {
             Map<String, Float> queryTokens = in.readMap(StreamInput::readString, StreamInput::readFloat);
             this.queryTokensSupplier = () -> queryTokens;
         }
+        // to be backward compatible with previous version, we need to use writeString/readString API instead of optionalString API
+        // after supporting query by tokens, queryText and modelId can be null. here we write an empty String instead
+        if (StringUtils.EMPTY.equals(this.queryText)) {
```

Comment:
Not getting it why can't using readOptionalString? I didn't see it breaks BWC, can you elaborate more?

Reply:
readString will directly readString from byte stream, while readOptionalString will first read a boolean from the byte Stream, then read a String.
Imagine we have 2 node, one is 2.14, the other is 2.13. In 2.13 we call writeString to stream, and in 2.14 we use readOptionalString to read it. This will break the serialization and deserialization process, making the behavior unpredictable. 

> I didn't see it breaks BWC
Because now we use a BWC-safe way to write/read it. Furthermore, an un-safe way will make bwc un-predictable, make it flaky test. But we can not make sure every API call will throw exception